src/Tribe/Service_Providers/Promoter_Connector.php

```diff
@@ -13,19 +13,29 @@ class Tribe__Service_Providers__Promoter_Connector extends tad_DI52_ServiceProvi
 	 * Binds and sets up implementations.
 	 */
 	public function register() {
-		tribe_singleton( 'common.promoter.view', 'Tribe__Promoter__View' );
-		tribe_singleton( 'common.promoter.rewrite', 'Tribe__Promoter__Rewrite' );
-		tribe_singleton( 'common.promoter.connector', 'Tribe__Promoter__Connector' );
 		tribe_singleton( 'common.promoter.auth', 'Tribe__Promoter__Auth' );
+		tribe_singleton( 'common.promoter.connector', 'Tribe__Promoter__Connector' );
+		tribe_singleton( 'common.promoter.pue', 'Tribe__Promoter__PUE', array( 'load' ) );
+		tribe_singleton( 'common.promoter.rewrite', 'Tribe__Promoter__Rewrite' );
+		tribe_singleton( 'common.promoter.view', 'Tribe__Promoter__View' );
 		$this->hook();
 	}
 
+	/**
+	 * Setup hooks for classes.
+	 */
 	private function hook() {
-		add_action( 'template_redirect', array( tribe( 'common.promoter.view' ), 'display_auth_check_view' ), 10, 0 );
-		add_action( 'tribe_common_promoter_pre_rewrite', array( tribe( 'common.promoter.rewrite' ), 'generate_core_rules' ) );
-		add_action( 'init', array( tribe( 'common.promoter.rewrite' ), 'add_rewrite_tags' ) );
-		add_filter( 'generate_rewrite_rules', array( tribe( 'common.promoter.rewrite' ), 'filter_generate' ) );
-		add_filter( 'rewrite_rules_array', array( tribe( 'common.promoter.rewrite' ), 'remove_percent_placeholders' ), 25 );
-		add_action( 'save_post', array( tribe( 'common.promoter.connector' ), 'notify_promoter_of_changes' ), 10, 1 );
+		$view      = tribe( 'common.promoter.view' );
+		$rewrite   = tribe( 'common.promoter.rewrite' );
+		$connector = tribe( 'common.promoter.connector' );
+
+		add_action( 'template_redirect', array( $view, 'display_auth_check_view' ), 10, 0 );
+		add_action( 'tribe_common_promoter_pre_rewrite', array( $rewrite, 'generate_core_rules' ) );
+		add_action( 'init', array( $rewrite, 'add_rewrite_tags' ) );
```

Comment:
I know these were merged already, but would love to see us use `tribe_callback( 'promoter.rewrite', 'add_rewrite_tags' );` 

Also, we are not namespacing `tribe-common`, it goes directly into the base `promoter.view` instead of `common.promoter.view`.

Reply:
This sounds like great feedback for the main branch code review, can you make those comments there? https://github.com/moderntribe/tribe-common/pull/902

Reply:
Actually I just made those changes for you